File: module4/unit3/db_mock.py
def mock_payment_database_query(user_id, status_code):
    """
    Simulates a database query for payment status.
    This mock assumes the database returns "PAID" if the status code is 200, 
    and "PENDING" otherwise.
    """
    if status_code == 200:
        return {"user_id": user_id, "payment_status": "PAID", "query_time_ms": 10}
    else:
        return {"user_id": user_id, "payment_status": "PENDING", "query_time_ms": 10}

File: module-4/unit3/typescript/src/dbMock.ts
export interface PaymentResponse {
    user_id: number;
    payment_status: "PAID" | "PENDING";
    query_time_ms: number;
}

/**
 * Simulates a database query for payment status.
 * Returns "PAID" if the status code is 200, and "PENDING" otherwise[.
 */
export function mockPaymentDatabaseQuery(userId: number, statusCode: number): PaymentResponse {
    if (statusCode === 200) {
        return { user_id: userId, payment_status: "PAID", query_time_ms: 10 };
    } else {
        return { user_id: userId, payment_status: "PENDING", query_time_ms: 10 };
    }
}